helpers/string.py

```diff
@@ -61,3 +63,47 @@ def replace(self, replacement_target):
                         string, replacement.output
                     )
         return replacement_target
+
+
+file_path_regex = regex.compile(
+    r"((\/*[\w\-]+\/)+([\w\.]+)(:\d+:\d+)*)",
+)
+
+
+def shorten_file_paths(string):
+    """
+        This function takes in a string and returns it with all the paths
+        it contains longer than 3 components shortened to 3 components
+
+        Example:
+            string = '''Error: expect(received).toBe(expected) // Object.is equality
+
+    Expected: 1
+    Received: -1
+        at Object.&lt;anonymous&gt; (/Users/josephsawaya/dev/test-result-action/demo/calculator/calculator.test.ts:10:31)
+        at Promise.then.completed (/Users/josephsawaya/dev/test-result-action/node_modules/jest-circus/build/utils.js:298:28)'''
+            shortened_string = shorten_file_paths(string)
+            print(shortened_string)
+
+            will print()
+
+    """
+
+    matches = file_path_regex.findall(string)
+    for match_tuple in matches:
+        file_path = match_tuple[0]
+        split_file_path = file_path.split("/")
+
+        # if the file_path has more than 3 components we should shorten it
+        if len(split_file_path) > 3:
```

Comment:
Thoughts on making this 3 a variable describing its purpose for readability? So something like MAX_COMPONENT_LENGTH or something like that?